lighthouse-viewer/app/src/lighthouse-report-viewer.js

```diff
@@ -35,16 +35,31 @@ class LighthouseReportViewer {
 
     this._dragAndDropper = new DragAndDrop(this._onFileLoad);
     this._github = new GithubApi();
+    this._psi = new PSIApi();
 
     /**
      * Used for tracking whether to offer to upload as a gist.
      * @type {boolean}
      */
     this._reportIsFromGist = false;
 
-    this._addEventListeners();
-    this._loadFromDeepLink();
-    this._listenForMessages();
+    this._reportIsFromPSI = false;
+
+    const params = new URLSearchParams(location.search);
+    const url = params.get('url');
+    if (url) {
```

Comment:
we should use something like `psi`, not the more generic `url`

Reply:
wb keep url, but require `provider=psi`?